### DISCUSSION TITLE
Failing to start Glances as a Service on Raspberry Pi 4

### DISCUSSION THREAD

**Discussion by simlocker** [nicolargo/glances#2524] [Q&A]
**Describe the bug**
Cannot start Glances as a service after **using the Standard Way** installation on a Raspberry Pi 4. 

**To Reproduce**
1. I created the glances.service in the /etc/systemd/system/ folder with these contents:

```
[Unit]
Description=Glances
After=network.target

[Service]
ExecStart=/usr/local/bin/glances -w
Restart=on-abort
RemainAfterExit=yes

[Install]
WantedBy=multi-user.target
```

Then I execute _sudo systemctl enable glances.service_ and finally start the service with _sudo systemctl enable glances.service_

**Expected behavior**

Service should be ready to start. However, it doesn't, and the following message appears on the service status:

```
● glances.service - Glances
     Loaded: loaded (/etc/systemd/system/glances.service; enabled; vendor preset: enabled)
     Active: failed (Result: exit-code) since Fri 2023-08-11 16:02:16 CEST; 1s ago
    Process: 64295 ExecStart=/usr/local/bin/glances -w (code=exited, status=1/FAILURE)
   Main PID: 64295 (code=exited, status=1/FAILURE)
        CPU: 64ms
```

**Environement (please complete the following information)**
Raspbery Pi 4 running Raspbian

Thank you.

Edit:

Not sure if it's actually related or not, but if I try to run Glances from pi user in this case, it works. If I try to run it being root, it fails:

```
**pi@raspi:/ $ glances --v**
Glances v3.4.0.3 with PsUtil v5.8.0
Log file: /home/pi/.local/share/glances/glances.log
pi@raspi:/ $ sudo su
**root@raspi:/# glances --v**
Traceback (most recent call last):
  File "/usr/local/bin/glances", line 5, in <module>
    from glances import main
ModuleNotFoundError: No module named 'glances'
```

**Comment by nicolargo** [comment]
Hi @gaston-arg 

yes it is related.

How do you install Glances on your system ?

**Comment by simlocker** [comment]
I used the Standard Way, so pip I guess.

**Reply by honmashinsei** [reply]
I have the same issue, running Glances as a service works fine, but when I want to run it as root (sudo glances) I get this:

`Traceback (most recent call last):
  File "/usr/local/bin/glances", line 5, in <module>
    from glances import main
ModuleNotFoundError: No module named 'glances'`

I installed through pip, but not sudo pip

**Comment by honmashinsei** [comment]
I was able to solve it through installing glances with root via pip, but I don't feel comfortable doing that, feels messy (I have two installs now).

("sudo pip install glances")

**Comment by baosws** [comment]
Have you tried adding `User=<your username>` to `[Service]` block in your service file to make it run the same python that was used to install glances? That worked to me.
